### PR TITLE
Polymer/multihop v6

### PR DESCRIPTION
**Pull request by alfredo-stonk**
## Description
- Refactor ibc-go multihop (v6) implementation so that multihop proof generation depends on abstract Endpoint instead of testing utils.
- Refactor multihop tests in line with existing ibc tests

### CODE REVIEW

File: modules/core/multihop/verify.go [hunk 1]
```diff
@@ -12,8 +12,62 @@ import (
...
+	value []byte,
+) error {
+	var proofs channeltypes.MsgMultihopProofs
+	if err := cdc.Unmarshal(proof, &proofs); err != nil {
+		return err
+	}
+
+	// check all connections are in OPEN state and that the connection IDs match and are in the right order
+	for i, connData := range proofs.ConnectionProofs {
+		//connData := proofs.ConnectionProofs[len(proofs.ConnectionProofs)-1-i]
```
Comment (dshiell) [inline modules/core/multihop/verify.go:32]: delete this comment?

File: testing/endpoint.go [hunk 2]
```diff
@@ -93,8 +93,16 @@ func (endpoint *Endpoint) CreateClient() (err error) {
 
 		height := endpoint.Counterparty.Chain.LastHeader.GetHeight().(clienttypes.Height)
 		clientState = ibctmtypes.NewClientState(
-			endpoint.Counterparty.Chain.ChainID, tmConfig.TrustLevel, tmConfig.TrustingPeriod, tmConfig.UnbondingPeriod, tmConfig.MaxClockDrift,
-			height, commitmenttypes.GetSDKSpecs(), UpgradePath, tmConfig.AllowUpdateAfterExpiry, tmConfig.AllowUpdateAfterMisbehaviour,
+			endpoint.Counterparty.Chain.ChainID,
```
Comment (dshiell) [inline testing/endpoint.go:96]: Do we want space reformatting on existing ibc-go files which are otherwise untouched?

Comment (dshiell) [reply testing/endpoint.go:96]: Mainly asking in case less changes helps with PR approval and less conflicts with other potential changes to endpoint.go

**Review by dshiell — APPROVED** [review]
Looks great overall. I had a couple small questions, but its another significant improvement overall!

File: testing/endpoint_multihop.go [hunk 3]
```diff
@@ -103,13 +231,140 @@ func (ep *EndpointM) SetChannelClosed() error {
...
+	packetKey := host.PacketAcknowledgementKey(packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence())
+	commitment := channeltypes.CommitAcknowledgement(ibcmock.MockAcknowledgement.Acknowledgement())
+	return ep.QueryMultihopProof(
+		packetKey, commitment,
+		fmt.Sprintf("packet acknowledgement: %s", packet.String()),
+	)
+}
+
+// QueryMultihopProof queries the proof for a key/value on this endpoint, which is verified on the counterparty chain.
+func (ep *EndpointM) QueryMultihopProof(key, expectedValue []byte, name string) []byte {
```
Comment (dshiell) [inline testing/endpoint_multihop.go:282]: Do we really need the expected Value here? Seems strange to require the expected value when querying?

Comment (dshiell) [reply testing/endpoint_multihop.go:282]: I was looking at `QueryProof()` which does not require a value which prompted this question.

File: modules/core/multihop/verify.go (hunk 1 above)
Comment (alfredo-stonk) [reply modules/core/multihop/verify.go:32]: done

File: testing/endpoint.go (hunk 2 above)
Comment (alfredo-stonk) [reply testing/endpoint.go:96]: good point. I didn't notice the format change

File: testing/endpoint_multihop.go (hunk 3 above)
Comment (alfredo-stonk) [reply testing/endpoint_multihop.go]: yea, makes sense. I dropped the expectedValue and made other relevant changes

**Comment by alfredo-stonk** [comment]
@dshiell addressed all comments, and added ChanPath.GenerateNonMembershipProof for your timeout PR. You can rebase yours off my branch

**Pull request merged**